api/v1/rpc_snap_refresh_status.go

```diff
@@ -0,0 +1,21 @@
+package apiv1
+
+// SnapRefreshRPC is the path for the SnapRefresh RPC.
+const SnapRefreshStatusRPC = "snap/refresh-status"
+
+// SnapRefreshStatusRequest is the request message for the SnapRefreshStatus RPC.
+type SnapRefreshStatusRequest struct {
+	// The change id belonging to a snap refresh/install operation.
+	ChangeID string `json:"changeId"`
+}
+
+// SnapRefreshStatusResponse is the response message for the SnapRefreshStatus RPC.
+type SnapRefreshStatusResponse struct {
+	// Status is the status of the snap refresh/install operation.
+	Status string `json:"status"`
+	// Completed is a boolean indicating if the snap refresh/install operation has completed.
+	// The status should be considered final when this is true.
+	Completed bool `json:"completed"`
```

Comment:
Why do we have an extra field for that? Can this not be part of Status?

Reply:
This is a mapping to the `Ready` field from the [snapd api](https://github.com/canonical/snapd/blob/a65eed3e1c1ab3b838ebf4aafb10741f894f6277/client/change.go#L31-L44). The status field represents the state of the operation currently, the `Ready` flag is marked as true once everything is complete (rolling back the snap on failure etc.).

Reply:
Makes sense, thanks!